Patches/DialogueBoxPatch.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace TOHE.Patches;
+[HarmonyPatch(typeof(DialogueBox))]
+internal class DialogueBoxPatch
+{
+    [HarmonyPatch(nameof(DialogueBox.Show)), HarmonyPostfix]
+    public static void Show_Postfix(DialogueBox __instance, [HarmonyArgument(0)]string dialouge)
+    {
+        if (dialouge.Contains("<size=0%>tohe</size>") && GameStates.IsInTask)
+        {
+            PlayerControl.LocalPlayer.ForceKillTimerContinue = true;
+        }
+    }
+
+    [HarmonyPatch(nameof(DialogueBox.Hide)), HarmonyPostfix]
+    public static void Hide_Postfix(DialogueBox __instance)
+    {
+        if (__instance.target.text.Contains("<size=0%>tohe</size>"))
+        {
+            PlayerControl.LocalPlayer.ForceKillTimerContinue = false;
+        }
+    }
```

Comment:
Wait, doesn't this lead to the fact that when closing cooldown will stop?

Reply:
> Wait, doesn't this lead to the fact that when closing cooldown will stop?

![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/139348239/8722208d-7f70-4bad-9778-cd896638ff20)
![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/139348239/6d4733fd-6374-41fa-94fb-b2aa2f57c3ea)

ForceKillTimerContinue is a single bool
The main decision should be done by iskilltimerenabled